docs/how-to/package-with-hatch.md

```diff
@@ -0,0 +1,385 @@
+
+```bash
+calculator
+```
+
+## Installing Hatch
+
+There are a few ways to install Hatch.
+See the [official docs on installation](https://hatch.pypa.io/latest/install/) for the best method for your operating system.
+
+Once installed, you should have the `hatch` command available on the command line.
+Run the following to check Hatch was installed correctly:
+
+```bash
+hatch
+```
+
+## Hatch new
+
+Hatch can create an initial directory structure and files with the `new` *subcommand*.
+Enter `hatch new` followed by the name of your project.
+For the calculator example, the name will be "textual calculator":
+
+```batch
+hatch new "textual calculator"
+```
+
+This will create the following directory structure:
+
+```
+textual-calculator
+├── LICENSE.txt
+├── README.md
+├── pyproject.toml
+├── src
+│   └── textual_calculator
+│       ├── __about__.py
+│       └── __init__.py
+└── tests
+    └── __init__.py
+```
+
+This follows a well established convention when packaging Python code, and will create the following files:
+
+- `LICENSE.txt` contains the license you want to distribute your code under.
+- `README.md` is a markdown file containing information about your project, which will be displayed in PyPi and Github (if you use it). You can edit this with information about your app and how to use it.
+- `pyproject.toml` is a [TOML](https://en.wikipedia.org/wiki/TOML) file which contains *metadata* (additional information) about your project and how to package it. This is a Python standard. This file may be edited manually or by a build tool (such as Hatch).
+- `src/textual_calculator/__about__.py` contains the version number of your app. You should update this when you release new versions.
+- `src/textual_calculator/__init__.py`  and `tests/__init__py` indicate the directory they are within contains Python code (these files are often empty).
+ 
+In the top level is a directory called `src`.
+This should contain a directory named after your project, and will be the name your code can be imported from.
+In our example, this directory is `textual_calculator` so we can do `import textual_calculator` in Python code.
+
+Additionally, there is a `tests` directory where you can add any [test](../guide/testing.md) code.
+
+### More on naming
+
+Note how Hatch replaced the space in the project name with a hyphen (i.e. `textual-calculator`), but the directory in `src` with an underscore (i.e. `textual_calculator`). This is because the directory in `src` contains the Python module, and a hyphen is not legal in a Python import. The top-level directory doesn't have this restriction and uses a hyphen, which is more typical for a directory name.
+
+Bear this in mind if your project name contains spaces.
+
+
+### Got existing code?
+
+The `hatch new` command assumes you are starting from scratch.
+If you have existing code you would like to package, navigate to your directory and run the following command:
+
+```
+hatch new --init <YOUR PROJECT NAME>
```

Comment:
It was [previously expressed](https://github.com/Textualize/textual/pull/3977#discussion_r1445964855) that we should avoid the use of `<...>` in documentation; I see it used a few times in here. Are we comfortable with this now?

Reply:
It is explained in the docs what the `<>` means, which I just added.